lecture_1/task/main.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+type jobApplicationResult struct {
+	Name   string
+	Age    int
+	Passed bool
+	Skills []string
+}
+
+const url = "https://run.mocky.io/v3/f7ceece5-47ee-4955-b974-438982267dc8"
+
+func getPassedApplications(list []jobApplicationResult) map[string]struct{} {
+	f, err := os.Create("passed_Go_Java.txt")
+	if err != nil {
+		log.Fatal(
+			errors.WithMessage(err, "opening a file"),
+		)
+	}
+
+	set := map[string]struct{}{}
+
+	for _, applicant := range list {
+		if applicant.Passed {
+			for _, skill := range applicant.Skills {
+				if skill == "Java" || skill == "Go" {
+					set[applicant.Name] = struct{}{}
+					f.WriteString(fmt.Sprint(applicant.Name) + "\n")
+					break
+				}
+			}
+		}
+	}
+
+	defer f.Close()
+	f.Sync()
+	return set
+}
+
+func getAllApplications(list []jobApplicationResult, passed map[string]struct{}) {
+	f, err := os.Create("skills.txt")
+	if err != nil {
+		log.Fatal(
+			errors.WithMessage(err, "opening a file"),
+		)
+	}
+
+	for _, applicant := range list {
+		_, found := passed[applicant.Name]
+		if found {
+			f.WriteString(fmt.Sprint(applicant.Name) + " - ")
+			for i, skill := range applicant.Skills {
+				if i+1 == len(applicant.Skills) {
+					f.WriteString(fmt.Sprint(skill) + "\n")
+				} else {
+					f.WriteString(fmt.Sprint(skill) + ", ")
+				}
+			}
+		}
+	}
+
+	defer f.Close()
+	f.Sync()
+}
+
+func main() {
```

Comment:
Code style recommendation: your main method has several levels of abstraction - you're opening an HTTP client and fetching some binary content, while in the last few lines you're calling some utility functions on a high-level. It would be preferred to stick to the same level of abstraction; in this specific case, having utility functions for all low-level operations (e.g. fetch HTTP response, unmarshal it into domain struct, filter out results, write results to file).